server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -2155,11 +2662,83 @@ public static Setting<TimeValue> timeSetting(
         return new Setting<>(
             simpleKey,
             s -> defaultValue.getStringRep(),
-            minMaxTimeValueParser(key, minValue, maxValue, isFiltered(properties)),
+            new MinMaxTimeValueParser(key, minValue, maxValue, isFiltered(properties)),
             properties
         );
     }
 
+    /**
+     * A writeable parser for time value have min and max value
+     *
+     */
+    public static class MinMaxTimeValueParser implements Function<String, TimeValue>, Writeable {
+        private String key;
+        private TimeValue minValue;
+        private TimeValue maxValue;
+        private boolean isFiltered;
+
+        public MinMaxTimeValueParser(String key, TimeValue minValue, TimeValue maxValue, boolean isFiltered) {
+            this.key = key;
+            this.minValue = minValue;
+            this.maxValue = maxValue;
+            this.isFiltered = isFiltered;
+        }
+
+        public MinMaxTimeValueParser(StreamInput in) throws IOException {
+            key = in.readString();
+            minValue = in.readTimeValue();
+            maxValue = in.readTimeValue();
+            isFiltered = in.readBoolean();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeString(key);
+            out.writeTimeValue(minValue);
+            out.writeTimeValue(maxValue);
+            out.writeBoolean(isFiltered);
+        }
+
+        // For testing
+        TimeValue getMin() {
+            return minValue;
+        }
+
+        // For testing
+        TimeValue getMax() {
+            return maxValue;
+        }
+
+        // For testing
+        String getKey() {
+            return key;
+        }
+
+        // For testing
+        boolean getFilterStatus() {
+            return isFiltered;
+        }
+
+        @Override
+        public TimeValue apply(String s) {
+            return minMaxTimeValueParser(key, minValue, maxValue, isFiltered).apply(s);
+        }
+
+        public boolean equals(Object obj) {
+            if (this == obj) return true;
+            if (obj == null || getClass() != obj.getClass()) return false;
+            MinMaxTimeValueParser that = (MinMaxTimeValueParser) obj;
+            return Objects.equals(key, that.key)
+                && Objects.equals(minValue, that.minValue)
+                && Objects.equals(maxValue, that.maxValue)
+                && Objects.equals(isFiltered, that.isFiltered);
+        }
+
+        public int hashCode() {
+            return Objects.hash(key, minValue, maxValue, isFiltered);
+        }
+    }
+
```

Comment:
This class looks like a little bulky here as we have added all the parsers in the same file? Can we create a separate class for the parsers and store them in one place?

Reply:
The purpose I add these parser class in setting class is try to avoid to create too much classes and yes, we can refactoring this class later. I can open another issue to work on it.

Reply:
We don't have to create multiple parser classes. Just one parser class which will consist of all the parsers.

Reply:
@mloufra can you link the issue here?